tests/lax_test.py

```diff
@@ -1338,54 +1338,58 @@ def testCumulativeReduce(self, op, onp_op, shape, dtype, axis, rng_factory):
     self._CheckAgainstNumpy(fun, onp_fun, args_maker)
 
   @parameterized.named_parameters(jtu.cases_from_list(
-      {"testcase_name": "_shape={}_axis={}".format(
-          jtu.format_shape_dtype_string(shape, dtype), axis),
-       "shape": shape, "dtype": dtype, "axis": axis}
+      {"testcase_name": "_shape={}_axis={}_isstable={}".format(
+          jtu.format_shape_dtype_string(shape, dtype), axis, is_stable),
+       "shape": shape, "dtype": dtype, "axis": axis, "is_stable": is_stable}
       for dtype in all_dtypes
       for shape in [(5,), (5, 7)]
-      for axis in [-1, len(shape) - 1]))
-  def testSort(self, shape, dtype, axis):
+      for axis in [-1, len(shape) - 1]
+      for is_stable in [False, True]))
+  def testSort(self, shape, dtype, axis, is_stable):
     # TODO(b/141131288): enable complex-valued sorts on TPU.
     if (onp.issubdtype(dtype, onp.complexfloating) and (
         (jtu.device_under_test() == "cpu" and jax.lib.version <= (0, 1, 47)) or
          jtu.device_under_test() == "tpu")):
       raise SkipTest("Complex-valued sort not implemented")
     rng = jtu.rand_default(self.rng())
     args_maker = lambda: [rng(shape, dtype)]
-    fun = lambda x: lax.sort(x, dimension=axis)
+    fun = lambda x: lax.sort(x, dimension=axis, is_stable=is_stable)
     self._CompileAndCheck(fun, args_maker)
 
   @parameterized.named_parameters(jtu.cases_from_list(
-      {"testcase_name": "_shape={}_axis={}".format(
-          jtu.format_shape_dtype_string(shape, dtype), axis),
-        "shape": shape, "dtype": dtype, "axis": axis}
+      {"testcase_name": "_shape={}_axis={}_isstable={}".format(
+          jtu.format_shape_dtype_string(shape, dtype), axis, is_stable),
+        "shape": shape, "dtype": dtype, "axis": axis, "is_stable": is_stable}
       for dtype in all_dtypes
       for shape in [(5,), (5, 7)]
-      for axis in [-1, len(shape) - 1]))
-  def testSortAgainstNumpy(self, shape, dtype, axis):
+      for axis in [-1, len(shape) - 1]
+      for is_stable in [False, True]))
+  def testSortAgainstNumpy(self, shape, dtype, axis, is_stable):
     # TODO(b/141131288): enable complex-valued sorts on TPU.
     if (onp.issubdtype(dtype, onp.complexfloating) and (
         (jtu.device_under_test() == "cpu" and jax.lib.version <= (0, 1, 47)) or
          jtu.device_under_test() == "tpu")):
       raise SkipTest("Complex-valued sort not implemented")
     rng = jtu.rand_default(self.rng())
     args_maker = lambda: [rng(shape, dtype)]
-    op = lambda x: lax.sort(x, dimension=axis)
-    numpy_op = lambda x: lax_reference.sort(x, axis)
+    op = lambda x: lax.sort(x, dimension=axis, is_stable=is_stable)
+    def numpy_op(x):
+      return lax_reference.sort(x, axis, kind='stable' if is_stable else None)
```

Comment:
This is causing issues in numpy 1.16; the traceback is:
```
Traceback (most recent call last):
  File "<embedded stdlib>/unittest/case.py", line 59, in testPartExecutor
    yield
  File "<embedded stdlib>/unittest/case.py", line 605, in run
    testMethod()
  File "/build/work/269db3e22bc8b0a4438c43b03951abcd1c0e/google3/runfiles/google3/third_party/py/absl/testing/parameterized.py", line 282, in bound_param_test
    return test_method(self, **testcase_params)
  File "/build/work/269db3e22bc8b0a4438c43b03951abcd1c0e/google3/runfiles/google3/third_party/py/jax/tests/lax_test.py", line 1378, in testSortAgainstNumpy
    self._CheckAgainstNumpy(op, numpy_op, args_maker)
  File "/build/work/269db3e22bc8b0a4438c43b03951abcd1c0e/google3/runfiles/google3/third_party/py/jax/test_util.py", line 838, in _CheckAgainstNumpy
    lax_ans = lax_op(*args)
  File "/build/work/269db3e22bc8b0a4438c43b03951abcd1c0e/google3/runfiles/google3/third_party/py/jax/tests/lax_test.py", line 1377, in numpy_op
    return lax_reference.sort(x, axis, kind='stable' if is_stable else None)
  File "/build/work/269db3e22bc8b0a4438c43b03951abcd1c0e/google3/runfiles/google3/third_party/py/numpy/core/fromnumeric.py", line 934, in sort
    a.sort(axis=axis, kind=kind, order=order)
TypeError: expected bytes, NoneType found
```

Reply:
Fixed. Sorry about that: is there a way to systematically test different versions of numpy besides creating an env for each one of them?

Reply:
No, I don't know of any other shortcut for that.